### DISCUSSION TITLE
Ubuntu 20.04 libstd++ is 10.3 - What to do with bit_cast ?

### DISCUSSION THREAD

**Discussion by jfalcou** [jfalcou/eve#1251] [Done & Done!]
The latest libstdc++ in the Ubuntu 20.04 channel is 10.3, and that doesn't have bit_cast. I'd definitely prefer eve not having that dependency so it will build on these common machines.

_Originally posted by @seanbaxter in https://github.com/jfalcou/eve/issues/1249#issuecomment-1046262971_

**Comment by jfalcou** [comment]
@seanbaxter We currently compiles with g++-11 at the minimum so I don't know if this boat already sailed anyway.

**Comment by DenisYaroshevskiy** [comment]
I am not too sure if libstdc++ is relevant for current emscripten. Seems like that uses libc++ at the moment.
When it comes to the compiler version we follow the latest official release until modules are a thing everywhere.
I think that was the policy.

**Comment by seanbaxter** [comment]
I don't use emscripten. But I'm on an updated 20.04 (which is still the current LTS release) and it only supports gcc-10.3. Is gcc-11 actually needed for language features? std::bit_cast is not a good reason to have a gcc-11 dependency.

**Comment by DenisYaroshevskiy** [comment]
I mean, do you use eve now? It requires bit_cast now.

**Reply by seanbaxter** [reply]
It's a big C++20 source base and i'm trying to bring up Circle compiler to compile it all. Bumping libstdc++ from 10.3 to 11.x represents another task I don't want to take right now.

**Reply by DenisYaroshevskiy** [reply]
We build with gcc-11 and clang-13 at the moment. And we might upgrade in some future.
If your goal is to get compiler to work, do you want to stick with a tagged version?

We might be able to revert some things if it's not too much hustle, but no guarantees, generally we stick to the latest compiler.

**Reply by jfalcou** [reply]
I think latest compiler is not a deal breaker, the issue is libstdc++ IIUC.
I don't think we use much of it apart from traits and concepts.

**Reply by seanbaxter** [reply]
Mm ya maybe libstdc++ is the issue.. The ranges stuff in particular seems to have been shifting with each version.

I can't just workaround stuff like bit_cast as it comes up. No big deal.

**Reply by jfalcou** [reply]
let's start fixing the TTS code then, that's trivial and remove this dependency. I think it helps right ?

**Comment by jfalcou** [comment]
Turns out the requirement on bit_cast is tied to TTS , our test system.
So we can actually fix that so we use memcpy if bit_cast is not available.

**Comment by jfalcou** [comment]
@seanbaxter THis should be fixed by #1255

**Reply by jfalcou** [reply]
which has been merged
